### PR TITLE
BEI-44: Unfavourite a recipe

### PR DESCRIPTION
**Pull request by LucianaDera**
WHY THIS PR?
We want users to have the ability to unfavourite a recipe.

WHAT DOES THIS PR DO?
It allows authenticated user to unfavourite a recipe.

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/core/controllers/favourite.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const unfavoriteRecipe = await favoriteRepository.findOne({
+      where:{
+        user_id: currentData.id,
+        recipe_id:findRecipe.id
+      }
+    })
+    if (unfavoriteRecipe) {
+      await unfavoriteRecipe.destroy();
+    }
+}
```
Comment (RICHYDOS) [inline src/core/controllers/favourite.ts:23]: add an extra line

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/controllers/favourite.ts (hunk 1 above)
Comment (LucianaDera) [reply src/core/controllers/favourite.ts:23]: theres an extra line

File: src/api/Route/favorite.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { authMiddleware } from '../middleware/authenticate';
+import * as RequestHandler from '../requestHandlers/favorite';
+
+const router = Router();
+
+
+router.delete('/:username/follow', authMiddleware, RequestHandler.unfollow);
+router.get('/:username', authMiddleware, RequestHandler.profile);
+
+export { router as profileRoutes };
```
Comment (RICHYDOS) [inline src/api/Route/favorite.ts:11]: what's going on here? I don't think this file should exist

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

**Review by hizikex — APPROVED** [review]

File: src/api/Route/recipe.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 import { createRecipe } from "../../core/controllers/recipe";
 import { authMiddleware } from "../middleware/authenticate";
-import * as RequestHandler from '../requestHandlers/comment';
+import * as RequestHandler from '../requestHandlers/favorite';
 
 const router = Router()
 
 router.post('/',authMiddleware, createRecipe);
-router.post('/:slug/comment', authMiddleware, RequestHandler.comment);
+router.delete('/:slug/favorite', authMiddleware, RequestHandler.unfavoriteHandler);
```
Comment (RICHYDOS) [inline src/api/Route/recipe.ts:9]: What happened to the create a comment route? Did you delete it ?

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/api/requestHandlers/comment.ts [hunk 4]
```diff
@@ -12,3 +12,7 @@ export const comment: RequestHandler = async (req, res, next): Promise<any> => {
     next(error);
   }
 };
+
+export function unfavoriteHandler(arg0: string, authMiddleware: (req: import("express").Request<import("express-serve-static-core").ParamsDictionary, any, any, import("qs").ParsedQs, Record<string, any>>, res: import("express").Response<any, Record<string, any>>, next: import("express").NextFunction) => Promise<...>, unfavoriteHandler: any) {
+  throw new Error("Function not implemented.");
+}
```
Comment (RICHYDOS) [inline src/api/requestHandlers/comment.ts:18]: This should not be here

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

**Review by RICHYDOS — APPROVED** [review]

**Review by RICHYDOS — APPROVED** [review]

**Review by hizikex — APPROVED** [review]

**Pull request merged**